lib/doc_handler.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+
+library dartpad.doc_handler;
+
+import 'dart:convert';
+import 'dart:html';
+import 'dart:math' as math;
+
+import 'dartservices_client/v1.dart';
+import 'editing/editor.dart';
+import 'context.dart';
+import 'services/common.dart';
+import 'src/ga.dart';
+
+import 'dart_pad.dart';
+import 'package:markd/markdown.dart' as markdown;
+
+class DocHandler {
+  Editor _editor;
+  Context _context;
+  DartservicesApi _dartServices;
```

Comment:
The dart services should be available as `dartServices` from the `dart_pad.dart` import; you won't need to pass it into this classes ctor.